site/en/docs/extensions/upgrading/to-service-workers/index.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: 'layouts/doc-post.njk'
+title: Upgrade to a service worker
+subhead: 'Replacing background or event pages with a service worker'
+description: 'A service worker enables extensions to run only when needed, saving resources.'
+date: 2023-03-08
+---
+
+A service worker replaces the extension's background or event page to ensure that background code stays off the main thread. This enables extensions to run only when needed, saving resources. 
+
+Background pages have been a fundamental component of extensions since their introduction. To put it simply, background pages provide an environment that lives independent of any other window or tab. This allows extensions to observe and act in response to events.
+
+## Differences between background scripts and extension service workers {: #differences-with-sws }
+
+In some contexts you'll see extension service workers called 'background scripts'. Although extension service workers do run in the background, calling them background scripts is somewhat misleading by implying identical capabilities. The differences are described below.
+
+### Changes from background pages {: #improvements-over-bg }
```

Comment:
@AmySteam I consulted with @ali-spivak. We retooled this a bit. (When my SW material is published, I'll reduce the whole difference section to a link.)